### PR TITLE
🖥️ Device check screen

### PR DESCRIPTION
**Pull request by marcoambrosini**

### CODE REVIEW

File: src/components/DeviceChecker/DeviceChecker.vue [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<template>
+	<Modal v-if="modal" size="prompt" @close="closeModal">
```
Comment (szaimen) [inline src/components/DeviceChecker/DeviceChecker.vue:23]: I fear the size `prompt` will not work as long as vue 5.0.0 is not released.

**Comment by marcoambrosini** [comment]
- [ ] microphone preview
- [ ] Title
- [ ] Modal size changes when switching devices

**Comment by nickvergessen** [comment]
- [ ] Default aspect ratio of the preview should either be 3:4 or 9:16

**Comment by nickvergessen** [comment]
- [ ] On the interstitial screen the icon is missing on the button or the padding should go

File: src/components/ConversationSettings/ConversationSettingsDialog.vue [hunk 2]
```diff
@@ -46,6 +46,14 @@
 			<NotificationsSettings :conversation="conversation" />
 		</AppSettingsSection>
 
+		<!-- Devices preview sceren -->
+		<AppSettingsSection
+			:title="t('spreed', 'Device check')">
+			<CheckboxRadioSwitch :checked.sync="showDeviceChecker">
+				{{ t('spreed',"Always show the device preview screen before joining a call in this conversation.") }}
```
Comment (nickvergessen) [inline src/components/ConversationSettings/ConversationSettingsDialog.vue:53]:
```suggestion
				{{ t('spreed', 'Always show the device preview screen before joining a call in this conversation.') }}
```

File: src/components/DeviceChecker/DeviceChecker.vue [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+						@update:deviceId="audioInputId = $event" />
+					<MediaDevicesSelector kind="videoinput"
+						:devices="devices"
+						:device-id="videoInputId"
+						@update:deviceId="videoInputId = $event" />
+				</template>
+			</div>
+			<CheckboxRadioSwitch :checked.sync="showDeviceChecker"
+				class="checkbox">
+				{{ t('spreed',"Always show this dialog before joining a call in this conversation.") }}
```
Comment (nickvergessen) [inline src/components/DeviceChecker/DeviceChecker.vue:129]:
```suggestion
				{{ t('spreed', 'Always show this dialog before joining a call in this conversation.') }}
```

File: src/components/MediaDevicesSelector.vue [hunk 4]
```diff
@@ -20,9 +20,15 @@
 
 <template>
 	<div class="media-devices-selector">
-		<h3 class="media-devices-selector__heading" :for="deviceSelectorId">
-			{{ deviceSelectorLabel }}
-		</h3>
+		<div class="media-devices-selector__icon">
+			<Microphone v-if="deviceIcon === 'microphone'"
+				title=""
```
Comment (nickvergessen) [inline src/components/MediaDevicesSelector.vue:25]: title or decorative?

File: src/components/MediaDevicesSelector.vue [hunk 5]
```diff
@@ -20,9 +20,15 @@
...
+				:size="16" />
+			<Video v-if="deviceIcon === 'camera'"
+				title=""
```
Comment (nickvergessen) [inline src/components/MediaDevicesSelector.vue:28]: title or decorative?

**Review by nickvergessen — APPROVED** [review]
Seems to work

**Pull request merged**